manim/mobject/types/vectorized_mobject.py

```diff
@@ -906,6 +906,181 @@ def __init__(self, *vmobjects, **kwargs):
+    show_keys : :class:`bool`
+            Whether to also display the key associated with 
+            the mobject. This might be useful when debugging,
+            especially when there are a lot of mobjects in the
+            :class:`VDict`. When displayed, the key is towards
+            the left of the mobject.
+            Defaults to False
+    submob_dict : :class:`dict`
+            Is the actual python dictionary that is used to bind
+            the keys to the mobjects
+            
+    '''
+
+    def __init__(self, *pairs, show_keys=False, **kwargs):
+        if not all([isinstance(m[1], VMobject) for m in pairs ]):
+            raise Exception("All submobjects must be of type VMobject")
+        VMobject.__init__(self, **kwargs)
+        self.show_keys = show_keys
+        self.submob_dict = {}
+        self.add(*pairs)
+        
+
+    def add(self, *pairs):
+        '''Adds the key-value pairs to the :class:`VDict` object.
+
+            Also, it internally adds the value to the `submobjects` :class:`list`
+            of :class:`~.Mobject`, which is responsible for actual on-screen display
+            
+            Parameters
+            ---------
+            pairs : Tuple[Hashable, :class:`~.VMobject`]
+                Each pair is a :class:`tuple` wherein the first 
+                element is the key for the mobject and the second
+                element is the actual mobject
+
+            Returns
+            -------
+            :class:`VDict`
+                Returns the :class:`VDict` object on which this method was called
+
+            Examples
+            --------
+            Normal usage::
+                square_obj = Square()
+                my_dict.add(('s', square_obj))
+            
+        '''
+        for pair in pairs:
+            key = pair[0]
+            value = pair[1]
+            
+            mob = value
+            if self.show_keys:
+                # This import is here and not at the top to avoid circular import
+                from ...mobject.svg.tex_mobject import TextMobject
+                key_text = TextMobject(str(key)).next_to(value, LEFT)
+                mob.add(key_text)
+            
+            self.submob_dict[key] = mob
+            super().add(value)
+        return self
+
+    def remove(self, key):
+        '''Removes the mobject from the :class:`VDict` object having the key `key`
+        
+            Also, it internally removes the mobject from the `submobjects` :class:`list`
+            of :class:`~.Mobject`, (which is responsible for removing it from the screen)
+
+            Parameters
+            ----------
+            key : Hashable
+                The key of the submoject to be removed
+
+            Returns
+            -------
+            :class:`VDict`
+                Returns the :class:`VDict` object on which this method was called
+
+            Examples
+            --------
+            Normal usage::
+            my_dict.remove('square')
```

Comment:
```suggestion
                my_dict.remove('square')
```

Reply:
On a side-note, is there some tool/library that helps with finding "style" errors/inconsistencies in code/documentation?

Reply:
for code yes, we've been discussing that on the autopep issues and stuff
for documentation, I'm not sure